utils/slice.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+// Join merges the provided slices into a single slice.
+//
+// TODO: Use slices.Concat once the minimum go version is 1.22.
```

Comment:
(no action required) Their signature is interesting:

```go
func Concat[S ~[]E, E any](slices ...S) S {
```

Without looking too deeply I guess it's to play nicely with other functions in that package.